src/datasets/containers.jl

```diff
@@ -26,6 +26,218 @@ function rglob(filepattern = "*", dir = pwd(), depth = 4)
     return vcat([glob(pattern, dir) for pattern in patterns[1:depth]]...)
 end
 
+function _ts2df(
+    full_file_path_and_name:: String,
+    replace_missing_vals_with:: String="NaN", # The value that missing values in the text file should be replaced with prior to parsing.
+)
+    "Load data from a .ts file into a DataFrames.jl DataFrame"
+
+    # Initialize flags and variables used when parsing the file
+    metadata_started = false 
+    data_started = false
+
+    has_problem_name_tag = false
+    has_timestamps_tag = false
+    has_univariate_tag = false
+    has_class_labels_tag = false
+    has_data_tag = false
+
+    previous_timestamp_was_int = nothing
+    prev_timestamp_was_timestamp = nothing
+    num_dimensions = nothing
+    is_first_case = true
+    instance_list = []
+    class_val_list = []
+    line_num = 0
+    series_length = 0
+
+    timestamps = false
+    class_labels = false
+
+    open(full_file_path_and_name, "r") do file
+        for ln in eachline(file)
+            # Strip white space from start/end of line and change to
+            # lowercase for use below
+            ln = lowercase(strip(ln, ' '))
+            # Empty lines are valid at any point in a file
+            if !isempty(ln)
+                # Check if this line contains metadata
+                # Please note that even though metadata is stored in this
+                # function it is not currently published externally
+                if startswith(ln, "@problemname")
+                    # Check that the associated value is valid
+                    tokens = split(ln, " ")
+                    token_len = length(tokens)
+
+                    # problem_name = line[len("@problemname") + 1:]
+                    has_problem_name_tag = true
+                    metadata_started = true
+
+                elseif startswith(ln, "@timestamps")
+                    # Check that the associated value is valid
+                    tokens = split(ln, " ")
+                    token_len = length(tokens)
+
+                    if tokens[2] == "true"
+                        timestamps = true
+                    else
+                        timestamps = false
+                    end
+
+                    has_timestamps_tag = true
+                    metadata_started = true
+
```

Comment:
There should be a way to extract out some of this logic. For now, having tests would help verify it's working.

Reply:
Yeah I think it could be extracted out for the current datasets we are using. I just copied it over from python library to get it working. 
By tests do you mean automated tests and creating a sample `.ts` file to run the test on ?

Reply:
Correct. You could also use a small existing one as long as the license is compatible.